link_test.go

```diff
@@ -124,6 +141,51 @@ func TestLinkFlowWithManualCreditor(t *testing.T) {
 	}
 }
 
+func TestLinkFlowWithDrain(t *testing.T) {
```

Comment:
Generally speaking, I think it would be preferable to write unit tests that use the fake as it's closer to a "real" scenario. But if this can be salvaged then that's good enough (but we should try not to add more in this vein).

Reply:
Agreed, I was already moving this one over to the newer style using a fakeConn.